### PR TITLE
[bsp/nuclei] Add BSP support for Nuclei HummingBird FPGA Evaluation Board 

### PR DESCRIPTION
**Pull request by fanghuaqi**
## 拉取/合并请求描述：(PR description)

* Add BSP support for [Nuclei HummingBird FPGA Evaluation Board 蜂鸟FPGA评估板](https://nucleisys.com/developboard.php)
   - Both 100T and 200T version are supported
   图片

* The Nuclei HummingBird FPGA Evaluation Board is used to evaluate Nuclei Commerical RISC-V Processor Core, such as Nuclei N/NX/UX classes, you can change the ``NUCLEI_SDK_CORE`` in ``rtconfig.py`` according to the CORE defined in your FPGA bitstream file, we support the following CORE list, see [Supported Nuclei Cores](https://doc.nucleisys.com/nuclei_sdk/develop/buildsystem.html#core)

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other style
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

File: bsp/nuclei/hbird_eval/SConscript [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# for module compiling
+import os
+Import('RTT_ROOT')
+from building import *
+
+cwd = GetCurrentDir()
+objs = []
+list = os.listdir(cwd)
+
+ASFLAGS = ' -I' + cwd
```
Comment (BernardXiong) [inline bsp/nuclei/hbird_eval/SConscript:10]: You can do it in a similar way in [qemu-vexpress-a9](https://github.com/RT-Thread/rt-thread/blob/master/bsp/qemu-vexpress-a9/SConstruct):
```python
env['ASCOM'] = env['ASPPCOM']
```

File: bsp/nuclei/hbird_eval/applications/main.c [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+ * Change Logs:
+ * Date           Author       Notes
+ * 2019-07-23     tyustli      first version
+ */
+
+#include <rtthread.h>
+#include <rtdevice.h>
+#include <nuclei_sdk_hal.h>
+
+#define THREAD_PRIORITY 19
```
Comment (BernardXiong) [inline bsp/nuclei/hbird_eval/applications/main.c:15]: better to keep the demo as simple as possible.

File: bsp/nuclei/libraries/hbird/HAL_Drivers/drv_uart.c [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+                                       &uart_obj[index]);
+        RT_ASSERT(result == RT_EOK);
+    }
+
+    return result;
+}
+
+void rt_hw_serial_thread_entry(void *parameter)
+{
+    struct hbird_uart_config *uart_cfg;
```
Comment (BernardXiong) [inline bsp/nuclei/libraries/hbird/HAL_Drivers/drv_uart.c:224]: What is the meaning of this thread & entry?

**Review by BernardXiong — CHANGES_REQUESTED** [review]
See comments, thanks.

File: bsp/nuclei/hbird_eval/SConscript (hunk 1 above)
Comment (fanghuaqi) [reply bsp/nuclei/hbird_eval/SConscript:10]: Hi @BernardXiong , I checked this file https://github.com/RT-Thread/rt-thread/blob/master/bsp/qemu-vexpress-a9/SConscript, and found I need to remove `ASFLAGS = ' -I' + cwd`, do you mean that?

File: bsp/nuclei/hbird_eval/applications/main.c (hunk 2 above)
Comment (fanghuaqi) [reply bsp/nuclei/hbird_eval/applications/main.c:15]: I can keep a simple demo.

File: bsp/nuclei/libraries/hbird/HAL_Drivers/drv_uart.c (hunk 3 above)
Comment (fanghuaqi) [reply bsp/nuclei/libraries/hbird/HAL_Drivers/drv_uart.c:224]: I use rtthread task to read uart input due to the uart irq line is not yet connected to core. 

This task `rt_hw_serial_thread_entry` is used in main.c

Comment (fanghuaqi) [reply bsp/nuclei/libraries/hbird/HAL_Drivers/drv_uart.c:224]: I changed the name to `rt_hw_serial_rcvtsk`

**Comment by fanghuaqi** [comment]
Just updated changes in a new commit to fixup the issues mentioned.

**Pull request merged**